src/cdk/css/css.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+ /* Parses arguments into a ngClass-like object */
+export function parseClassList(classList:
```

Comment:
I'm sure whether we need to support all of these formats. `NgClass` has them, presumably, because the expressions are dynamic and there a lot of ways to combine static classes and dynamic ones. In our case we only add the classes on init and we don't care about updating them, which can be done just as easily with an array.

Reply:
The original ticket expressed desire to standardize this format. Ref: #6372

Reply:
FWIW NgClass seemed like a natural thing to support for template-based overlays like select, menu, autocomplete, tooltip. For dialog and snackbar, it's probably less important, but there are some good use cases for having the ability to dynamically update classes, https://github.com/angular/material2/issues/6206#issuecomment-319528256.

Further, I would personally find an NgClass cdk utility useful for custom overlay-based components!

Reply:
Template-based overlays should "just work" as they are at the moment (you can stick an `NgClass` on an element inside the `TemplateRef`). This PR adds a similar signature to the programmatic API, but it's not _really_ `NgClass` because it won't update dynamically.

Reply:
Ok tbh, I hadn't actually read the pr 😄 . My initial concern in filing #6372 was how snackbar and dialog differed in support and naming. Combined with little type info available in the docs (as https://github.com/angular/material2/pull/6947 is trying to solve) it's hard to anticipate exactly what will happen just by looking at the docs.

I agree, `{[klass: string]: any}` is probably extraneous (and maybe misrepresenting) if classes don't update dynamically. While I still think that's a valid feature via #6206, maybe for another time.

Reply:
Ya, so basically this PR:

- Unifies all the css class names to `panelClass`
- Normalizes all the signatures to `ngClass`-like

That means you can pass it an string, string array or object key/value pair. In the cases of the snackbar and overlay, there is no `ngClass` used so it just does a evaluate of the condition and matches it. I personally think the consistent signature is valuable and in the situation where there isn't real `ngClass` I could do an actual implementation I just didn't do it because I didn't know how far we wanted to take this.

@jelbourn - Can you advise on next steps with this? Currently the PR is just in a evaluation phase and I know you expressed concerns with adding another CDK lib here too so I didn't want to take it too far without first review.